diff -ur ./zlib_source_original/CMakeLists.txt ./zlib_source/CMakeLists.txt
--- ./zlib_source_original/CMakeLists.txt	2010-03-28 22:06:13.000000000 +0300
+++ ./zlib_source/CMakeLists.txt	2017-08-18 09:46:38.330696800 +0300
@@ -7,6 +7,21 @@
     option(BUILD_SHARED_LIBS "Build a shared library form of zlib" ON)
 endif()
 
+if (MSVC)
+    # setup static linking with crt
+    set(CompilerFlags
+            CMAKE_CXX_FLAGS
+            CMAKE_CXX_FLAGS_DEBUG
+            CMAKE_CXX_FLAGS_RELEASE
+            CMAKE_C_FLAGS
+            CMAKE_C_FLAGS_DEBUG
+            CMAKE_C_FLAGS_RELEASE
+    )
+    foreach(CompilerFlag ${CompilerFlags})
+        string(REPLACE "/MD" "/MT" ${CompilerFlag} "${${CompilerFlag}}")
+    endforeach()
+endif()
+
 include(CheckTypeSize)
 include(CheckFunctionExists)
 include(CheckIncludeFile)
