diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile	2018-03-19 11:25:20.000000000 +0300
@@ -6,9 +6,9 @@
 OBJS_DIR  = build
 RMDIR     = rm -fr
 ECHO      = echo
-CCLD      =  g++
-CXX       =  g++
-CC        =  gcc
+CCLD      =  clang++
+CXX       =  clang++
+CC        =  clang
 RANLIB    = ranlib
 AR		 = ar
 STRIP     = strip
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevel.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevel.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevel.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevel.mk	2018-03-19 11:36:17.000000000 +0300
@@ -93,11 +93,12 @@
 LowLevel_debug_common_cflags    += $(addprefix -D, $(LowLevel_debug_defines))
 LowLevel_debug_common_cflags    += $(addprefix -I, $(LowLevel_debug_hpaths))
 LowLevel_debug_common_cflags  += -m64
-LowLevel_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevel_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevel_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevel_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevel_debug_common_cflags  += -Wno-missing-field-initializers
 LowLevel_debug_common_cflags  += -g3 -gdwarf-2
+LowLevel_debug_common_cflags  += $(DAVA_custom_flags)
 LowLevel_debug_cflags	:= $(LowLevel_debug_common_cflags)
 LowLevel_debug_cppflags	:= $(LowLevel_debug_common_cflags)
 LowLevel_debug_lflags    := $(LowLevel_custom_lflags)
@@ -208,11 +209,12 @@
 LowLevel_checked_common_cflags    += $(addprefix -D, $(LowLevel_checked_defines))
 LowLevel_checked_common_cflags    += $(addprefix -I, $(LowLevel_checked_hpaths))
 LowLevel_checked_common_cflags  += -m64
-LowLevel_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevel_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevel_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevel_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevel_checked_common_cflags  += -Wno-missing-field-initializers
 LowLevel_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+LowLevel_checked_common_cflags  += $(DAVA_custom_flags)
 LowLevel_checked_cflags	:= $(LowLevel_checked_common_cflags)
 LowLevel_checked_cppflags	:= $(LowLevel_checked_common_cflags)
 LowLevel_checked_lflags    := $(LowLevel_custom_lflags)
@@ -323,11 +325,12 @@
 LowLevel_profile_common_cflags    += $(addprefix -D, $(LowLevel_profile_defines))
 LowLevel_profile_common_cflags    += $(addprefix -I, $(LowLevel_profile_hpaths))
 LowLevel_profile_common_cflags  += -m64
-LowLevel_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevel_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevel_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevel_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevel_profile_common_cflags  += -Wno-missing-field-initializers
 LowLevel_profile_common_cflags  += -O3 -fno-strict-aliasing
+LowLevel_profile_common_cflags  += $(DAVA_custom_flags)
 LowLevel_profile_cflags	:= $(LowLevel_profile_common_cflags)
 LowLevel_profile_cppflags	:= $(LowLevel_profile_common_cflags)
 LowLevel_profile_lflags    := $(LowLevel_custom_lflags)
@@ -437,11 +440,12 @@
 LowLevel_release_common_cflags    += $(addprefix -D, $(LowLevel_release_defines))
 LowLevel_release_common_cflags    += $(addprefix -I, $(LowLevel_release_hpaths))
 LowLevel_release_common_cflags  += -m64
-LowLevel_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevel_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevel_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevel_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevel_release_common_cflags  += -Wno-missing-field-initializers
 LowLevel_release_common_cflags  += -O3 -fno-strict-aliasing
+LowLevel_release_common_cflags  += $(DAVA_custom_flags)
 LowLevel_release_cflags	:= $(LowLevel_release_common_cflags)
 LowLevel_release_cppflags	:= $(LowLevel_release_common_cflags)
 LowLevel_release_lflags    := $(LowLevel_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelAABB.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelAABB.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelAABB.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelAABB.mk	2018-03-19 11:36:17.000000000 +0300
@@ -64,11 +64,12 @@
 LowLevelAABB_debug_common_cflags    += $(addprefix -D, $(LowLevelAABB_debug_defines))
 LowLevelAABB_debug_common_cflags    += $(addprefix -I, $(LowLevelAABB_debug_hpaths))
 LowLevelAABB_debug_common_cflags  += -m64
-LowLevelAABB_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelAABB_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelAABB_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelAABB_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelAABB_debug_common_cflags  += -Wno-missing-field-initializers
 LowLevelAABB_debug_common_cflags  += -g3 -gdwarf-2
+LowLevelAABB_debug_common_cflags  += $(DAVA_custom_flags)
 LowLevelAABB_debug_cflags	:= $(LowLevelAABB_debug_common_cflags)
 LowLevelAABB_debug_cppflags	:= $(LowLevelAABB_debug_common_cflags)
 LowLevelAABB_debug_lflags    := $(LowLevelAABB_custom_lflags)
@@ -164,11 +165,12 @@
 LowLevelAABB_checked_common_cflags    += $(addprefix -D, $(LowLevelAABB_checked_defines))
 LowLevelAABB_checked_common_cflags    += $(addprefix -I, $(LowLevelAABB_checked_hpaths))
 LowLevelAABB_checked_common_cflags  += -m64
-LowLevelAABB_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelAABB_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelAABB_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelAABB_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelAABB_checked_common_cflags  += -Wno-missing-field-initializers
 LowLevelAABB_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+LowLevelAABB_checked_common_cflags  += $(DAVA_custom_flags)
 LowLevelAABB_checked_cflags	:= $(LowLevelAABB_checked_common_cflags)
 LowLevelAABB_checked_cppflags	:= $(LowLevelAABB_checked_common_cflags)
 LowLevelAABB_checked_lflags    := $(LowLevelAABB_custom_lflags)
@@ -264,11 +266,12 @@
 LowLevelAABB_profile_common_cflags    += $(addprefix -D, $(LowLevelAABB_profile_defines))
 LowLevelAABB_profile_common_cflags    += $(addprefix -I, $(LowLevelAABB_profile_hpaths))
 LowLevelAABB_profile_common_cflags  += -m64
-LowLevelAABB_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelAABB_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelAABB_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelAABB_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelAABB_profile_common_cflags  += -Wno-missing-field-initializers
 LowLevelAABB_profile_common_cflags  += -O3 -fno-strict-aliasing
+LowLevelAABB_profile_common_cflags  += $(DAVA_custom_flags)
 LowLevelAABB_profile_cflags	:= $(LowLevelAABB_profile_common_cflags)
 LowLevelAABB_profile_cppflags	:= $(LowLevelAABB_profile_common_cflags)
 LowLevelAABB_profile_lflags    := $(LowLevelAABB_custom_lflags)
@@ -363,11 +366,12 @@
 LowLevelAABB_release_common_cflags    += $(addprefix -D, $(LowLevelAABB_release_defines))
 LowLevelAABB_release_common_cflags    += $(addprefix -I, $(LowLevelAABB_release_hpaths))
 LowLevelAABB_release_common_cflags  += -m64
-LowLevelAABB_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelAABB_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelAABB_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelAABB_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelAABB_release_common_cflags  += -Wno-missing-field-initializers
 LowLevelAABB_release_common_cflags  += -O3 -fno-strict-aliasing
+LowLevelAABB_release_common_cflags  += $(DAVA_custom_flags)
 LowLevelAABB_release_cflags	:= $(LowLevelAABB_release_common_cflags)
 LowLevelAABB_release_cppflags	:= $(LowLevelAABB_release_common_cflags)
 LowLevelAABB_release_lflags    := $(LowLevelAABB_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelCloth.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelCloth.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelCloth.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelCloth.mk	2018-03-19 11:36:17.000000000 +0300
@@ -60,11 +60,12 @@
 LowLevelCloth_debug_common_cflags    += $(addprefix -D, $(LowLevelCloth_debug_defines))
 LowLevelCloth_debug_common_cflags    += $(addprefix -I, $(LowLevelCloth_debug_hpaths))
 LowLevelCloth_debug_common_cflags  += -m64
-LowLevelCloth_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelCloth_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelCloth_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelCloth_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelCloth_debug_common_cflags  += -Wno-missing-field-initializers
 LowLevelCloth_debug_common_cflags  += -g3 -gdwarf-2
+LowLevelCloth_debug_common_cflags  += $(DAVA_custom_flags)
 LowLevelCloth_debug_cflags	:= $(LowLevelCloth_debug_common_cflags)
 LowLevelCloth_debug_cppflags	:= $(LowLevelCloth_debug_common_cflags)
 LowLevelCloth_debug_lflags    := $(LowLevelCloth_custom_lflags)
@@ -150,11 +151,12 @@
 LowLevelCloth_checked_common_cflags    += $(addprefix -D, $(LowLevelCloth_checked_defines))
 LowLevelCloth_checked_common_cflags    += $(addprefix -I, $(LowLevelCloth_checked_hpaths))
 LowLevelCloth_checked_common_cflags  += -m64
-LowLevelCloth_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelCloth_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelCloth_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelCloth_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelCloth_checked_common_cflags  += -Wno-missing-field-initializers
 LowLevelCloth_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+LowLevelCloth_checked_common_cflags  += $(DAVA_custom_flags)
 LowLevelCloth_checked_cflags	:= $(LowLevelCloth_checked_common_cflags)
 LowLevelCloth_checked_cppflags	:= $(LowLevelCloth_checked_common_cflags)
 LowLevelCloth_checked_lflags    := $(LowLevelCloth_custom_lflags)
@@ -240,11 +242,12 @@
 LowLevelCloth_profile_common_cflags    += $(addprefix -D, $(LowLevelCloth_profile_defines))
 LowLevelCloth_profile_common_cflags    += $(addprefix -I, $(LowLevelCloth_profile_hpaths))
 LowLevelCloth_profile_common_cflags  += -m64
-LowLevelCloth_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelCloth_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelCloth_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelCloth_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelCloth_profile_common_cflags  += -Wno-missing-field-initializers
 LowLevelCloth_profile_common_cflags  += -O3 -fno-strict-aliasing
+LowLevelCloth_profile_common_cflags  += $(DAVA_custom_flags)
 LowLevelCloth_profile_cflags	:= $(LowLevelCloth_profile_common_cflags)
 LowLevelCloth_profile_cppflags	:= $(LowLevelCloth_profile_common_cflags)
 LowLevelCloth_profile_lflags    := $(LowLevelCloth_custom_lflags)
@@ -329,11 +332,12 @@
 LowLevelCloth_release_common_cflags    += $(addprefix -D, $(LowLevelCloth_release_defines))
 LowLevelCloth_release_common_cflags    += $(addprefix -I, $(LowLevelCloth_release_hpaths))
 LowLevelCloth_release_common_cflags  += -m64
-LowLevelCloth_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelCloth_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelCloth_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelCloth_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelCloth_release_common_cflags  += -Wno-missing-field-initializers
 LowLevelCloth_release_common_cflags  += -O3 -fno-strict-aliasing
+LowLevelCloth_release_common_cflags  += $(DAVA_custom_flags)
 LowLevelCloth_release_cflags	:= $(LowLevelCloth_release_common_cflags)
 LowLevelCloth_release_cppflags	:= $(LowLevelCloth_release_common_cflags)
 LowLevelCloth_release_lflags    := $(LowLevelCloth_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelDynamics.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelDynamics.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelDynamics.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelDynamics.mk	2018-03-19 11:36:17.000000000 +0300
@@ -86,11 +86,12 @@
 LowLevelDynamics_debug_common_cflags    += $(addprefix -D, $(LowLevelDynamics_debug_defines))
 LowLevelDynamics_debug_common_cflags    += $(addprefix -I, $(LowLevelDynamics_debug_hpaths))
 LowLevelDynamics_debug_common_cflags  += -m64
-LowLevelDynamics_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelDynamics_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelDynamics_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelDynamics_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelDynamics_debug_common_cflags  += -Wno-missing-field-initializers
 LowLevelDynamics_debug_common_cflags  += -g3 -gdwarf-2
+LowLevelDynamics_debug_common_cflags  += $(DAVA_custom_flags)
 LowLevelDynamics_debug_cflags	:= $(LowLevelDynamics_debug_common_cflags)
 LowLevelDynamics_debug_cppflags	:= $(LowLevelDynamics_debug_common_cflags)
 LowLevelDynamics_debug_lflags    := $(LowLevelDynamics_custom_lflags)
@@ -191,11 +192,12 @@
 LowLevelDynamics_checked_common_cflags    += $(addprefix -D, $(LowLevelDynamics_checked_defines))
 LowLevelDynamics_checked_common_cflags    += $(addprefix -I, $(LowLevelDynamics_checked_hpaths))
 LowLevelDynamics_checked_common_cflags  += -m64
-LowLevelDynamics_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelDynamics_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelDynamics_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelDynamics_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelDynamics_checked_common_cflags  += -Wno-missing-field-initializers
 LowLevelDynamics_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+LowLevelDynamics_checked_common_cflags  += $(DAVA_custom_flags)
 LowLevelDynamics_checked_cflags	:= $(LowLevelDynamics_checked_common_cflags)
 LowLevelDynamics_checked_cppflags	:= $(LowLevelDynamics_checked_common_cflags)
 LowLevelDynamics_checked_lflags    := $(LowLevelDynamics_custom_lflags)
@@ -296,11 +298,12 @@
 LowLevelDynamics_profile_common_cflags    += $(addprefix -D, $(LowLevelDynamics_profile_defines))
 LowLevelDynamics_profile_common_cflags    += $(addprefix -I, $(LowLevelDynamics_profile_hpaths))
 LowLevelDynamics_profile_common_cflags  += -m64
-LowLevelDynamics_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelDynamics_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelDynamics_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelDynamics_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelDynamics_profile_common_cflags  += -Wno-missing-field-initializers
 LowLevelDynamics_profile_common_cflags  += -O3 -fno-strict-aliasing
+LowLevelDynamics_profile_common_cflags  += $(DAVA_custom_flags)
 LowLevelDynamics_profile_cflags	:= $(LowLevelDynamics_profile_common_cflags)
 LowLevelDynamics_profile_cppflags	:= $(LowLevelDynamics_profile_common_cflags)
 LowLevelDynamics_profile_lflags    := $(LowLevelDynamics_custom_lflags)
@@ -400,11 +403,12 @@
 LowLevelDynamics_release_common_cflags    += $(addprefix -D, $(LowLevelDynamics_release_defines))
 LowLevelDynamics_release_common_cflags    += $(addprefix -I, $(LowLevelDynamics_release_hpaths))
 LowLevelDynamics_release_common_cflags  += -m64
-LowLevelDynamics_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelDynamics_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelDynamics_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelDynamics_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelDynamics_release_common_cflags  += -Wno-missing-field-initializers
 LowLevelDynamics_release_common_cflags  += -O3 -fno-strict-aliasing
+LowLevelDynamics_release_common_cflags  += $(DAVA_custom_flags)
 LowLevelDynamics_release_cflags	:= $(LowLevelDynamics_release_common_cflags)
 LowLevelDynamics_release_cppflags	:= $(LowLevelDynamics_release_common_cflags)
 LowLevelDynamics_release_lflags    := $(LowLevelDynamics_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelParticles.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelParticles.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelParticles.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.LowLevelParticles.mk	2018-03-19 11:36:17.000000000 +0300
@@ -73,11 +73,12 @@
 LowLevelParticles_debug_common_cflags    += $(addprefix -D, $(LowLevelParticles_debug_defines))
 LowLevelParticles_debug_common_cflags    += $(addprefix -I, $(LowLevelParticles_debug_hpaths))
 LowLevelParticles_debug_common_cflags  += -m64
-LowLevelParticles_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelParticles_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelParticles_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelParticles_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelParticles_debug_common_cflags  += -Wno-missing-field-initializers
 LowLevelParticles_debug_common_cflags  += -g3 -gdwarf-2
+LowLevelParticles_debug_common_cflags  += $(DAVA_custom_flags)
 LowLevelParticles_debug_cflags	:= $(LowLevelParticles_debug_common_cflags)
 LowLevelParticles_debug_cppflags	:= $(LowLevelParticles_debug_common_cflags)
 LowLevelParticles_debug_lflags    := $(LowLevelParticles_custom_lflags)
@@ -172,11 +173,12 @@
 LowLevelParticles_checked_common_cflags    += $(addprefix -D, $(LowLevelParticles_checked_defines))
 LowLevelParticles_checked_common_cflags    += $(addprefix -I, $(LowLevelParticles_checked_hpaths))
 LowLevelParticles_checked_common_cflags  += -m64
-LowLevelParticles_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelParticles_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelParticles_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelParticles_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelParticles_checked_common_cflags  += -Wno-missing-field-initializers
 LowLevelParticles_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+LowLevelParticles_checked_common_cflags  += $(DAVA_custom_flags)
 LowLevelParticles_checked_cflags	:= $(LowLevelParticles_checked_common_cflags)
 LowLevelParticles_checked_cppflags	:= $(LowLevelParticles_checked_common_cflags)
 LowLevelParticles_checked_lflags    := $(LowLevelParticles_custom_lflags)
@@ -271,11 +273,12 @@
 LowLevelParticles_profile_common_cflags    += $(addprefix -D, $(LowLevelParticles_profile_defines))
 LowLevelParticles_profile_common_cflags    += $(addprefix -I, $(LowLevelParticles_profile_hpaths))
 LowLevelParticles_profile_common_cflags  += -m64
-LowLevelParticles_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelParticles_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelParticles_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelParticles_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelParticles_profile_common_cflags  += -Wno-missing-field-initializers
 LowLevelParticles_profile_common_cflags  += -O3 -fno-strict-aliasing
+LowLevelParticles_profile_common_cflags  += $(DAVA_custom_flags)
 LowLevelParticles_profile_cflags	:= $(LowLevelParticles_profile_common_cflags)
 LowLevelParticles_profile_cppflags	:= $(LowLevelParticles_profile_common_cflags)
 LowLevelParticles_profile_lflags    := $(LowLevelParticles_custom_lflags)
@@ -369,11 +372,12 @@
 LowLevelParticles_release_common_cflags    += $(addprefix -D, $(LowLevelParticles_release_defines))
 LowLevelParticles_release_common_cflags    += $(addprefix -I, $(LowLevelParticles_release_hpaths))
 LowLevelParticles_release_common_cflags  += -m64
-LowLevelParticles_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+LowLevelParticles_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 LowLevelParticles_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 LowLevelParticles_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 LowLevelParticles_release_common_cflags  += -Wno-missing-field-initializers
 LowLevelParticles_release_common_cflags  += -O3 -fno-strict-aliasing
+LowLevelParticles_release_common_cflags  += $(DAVA_custom_flags)
 LowLevelParticles_release_cflags	:= $(LowLevelParticles_release_common_cflags)
 LowLevelParticles_release_cppflags	:= $(LowLevelParticles_release_common_cflags)
 LowLevelParticles_release_lflags    := $(LowLevelParticles_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PhysX.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PhysX.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PhysX.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PhysX.mk	2018-03-19 11:36:17.000000000 +0300
@@ -146,11 +146,12 @@
 PhysX_release_common_cflags    += $(addprefix -D, $(PhysX_release_defines))
 PhysX_release_common_cflags    += $(addprefix -I, $(PhysX_release_hpaths))
 PhysX_release_common_cflags  += -m64
-PhysX_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysX_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysX_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysX_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysX_release_common_cflags  += -Wno-missing-field-initializers
 PhysX_release_common_cflags  += -O3 -fno-strict-aliasing
+PhysX_release_common_cflags  += $(DAVA_custom_flags)
 PhysX_release_cflags	:= $(PhysX_release_common_cflags)
 PhysX_release_cppflags	:= $(PhysX_release_common_cflags)
 PhysX_release_lflags    := $(PhysX_custom_lflags)
@@ -291,11 +292,12 @@
 PhysX_debug_common_cflags    += $(addprefix -D, $(PhysX_debug_defines))
 PhysX_debug_common_cflags    += $(addprefix -I, $(PhysX_debug_hpaths))
 PhysX_debug_common_cflags  += -m64
-PhysX_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysX_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysX_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysX_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysX_debug_common_cflags  += -Wno-missing-field-initializers
 PhysX_debug_common_cflags  += -g3 -gdwarf-2
+PhysX_debug_common_cflags  += $(DAVA_custom_flags)
 PhysX_debug_cflags	:= $(PhysX_debug_common_cflags)
 PhysX_debug_cppflags	:= $(PhysX_debug_common_cflags)
 PhysX_debug_lflags    := $(PhysX_custom_lflags)
@@ -435,11 +437,12 @@
 PhysX_checked_common_cflags    += $(addprefix -D, $(PhysX_checked_defines))
 PhysX_checked_common_cflags    += $(addprefix -I, $(PhysX_checked_hpaths))
 PhysX_checked_common_cflags  += -m64
-PhysX_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysX_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysX_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysX_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysX_checked_common_cflags  += -Wno-missing-field-initializers
 PhysX_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+PhysX_checked_common_cflags  += $(DAVA_custom_flags)
 PhysX_checked_cflags	:= $(PhysX_checked_common_cflags)
 PhysX_checked_cppflags	:= $(PhysX_checked_common_cflags)
 PhysX_checked_lflags    := $(PhysX_custom_lflags)
@@ -579,11 +582,12 @@
 PhysX_profile_common_cflags    += $(addprefix -D, $(PhysX_profile_defines))
 PhysX_profile_common_cflags    += $(addprefix -I, $(PhysX_profile_hpaths))
 PhysX_profile_common_cflags  += -m64
-PhysX_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysX_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysX_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysX_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysX_profile_common_cflags  += -Wno-missing-field-initializers
 PhysX_profile_common_cflags  += -O3 -fno-strict-aliasing
+PhysX_profile_common_cflags  += $(DAVA_custom_flags)
 PhysX_profile_cflags	:= $(PhysX_profile_common_cflags)
 PhysX_profile_cppflags	:= $(PhysX_profile_common_cflags)
 PhysX_profile_lflags    := $(PhysX_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXCharacterKinematic.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXCharacterKinematic.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXCharacterKinematic.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXCharacterKinematic.mk	2018-03-19 11:36:17.000000000 +0300
@@ -75,11 +75,12 @@
 PhysXCharacterKinematic_debug_common_cflags    += $(addprefix -D, $(PhysXCharacterKinematic_debug_defines))
 PhysXCharacterKinematic_debug_common_cflags    += $(addprefix -I, $(PhysXCharacterKinematic_debug_hpaths))
 PhysXCharacterKinematic_debug_common_cflags  += -m64
-PhysXCharacterKinematic_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXCharacterKinematic_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXCharacterKinematic_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXCharacterKinematic_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXCharacterKinematic_debug_common_cflags  += -Wno-missing-field-initializers
 PhysXCharacterKinematic_debug_common_cflags  += -g3 -gdwarf-2
+PhysXCharacterKinematic_debug_common_cflags  += $(DAVA_custom_flags)
 PhysXCharacterKinematic_debug_cflags	:= $(PhysXCharacterKinematic_debug_common_cflags)
 PhysXCharacterKinematic_debug_cppflags	:= $(PhysXCharacterKinematic_debug_common_cflags)
 PhysXCharacterKinematic_debug_lflags    := $(PhysXCharacterKinematic_custom_lflags)
@@ -183,11 +184,12 @@
 PhysXCharacterKinematic_checked_common_cflags    += $(addprefix -D, $(PhysXCharacterKinematic_checked_defines))
 PhysXCharacterKinematic_checked_common_cflags    += $(addprefix -I, $(PhysXCharacterKinematic_checked_hpaths))
 PhysXCharacterKinematic_checked_common_cflags  += -m64
-PhysXCharacterKinematic_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXCharacterKinematic_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXCharacterKinematic_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXCharacterKinematic_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXCharacterKinematic_checked_common_cflags  += -Wno-missing-field-initializers
 PhysXCharacterKinematic_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+PhysXCharacterKinematic_checked_common_cflags  += $(DAVA_custom_flags)
 PhysXCharacterKinematic_checked_cflags	:= $(PhysXCharacterKinematic_checked_common_cflags)
 PhysXCharacterKinematic_checked_cppflags	:= $(PhysXCharacterKinematic_checked_common_cflags)
 PhysXCharacterKinematic_checked_lflags    := $(PhysXCharacterKinematic_custom_lflags)
@@ -291,11 +293,12 @@
 PhysXCharacterKinematic_profile_common_cflags    += $(addprefix -D, $(PhysXCharacterKinematic_profile_defines))
 PhysXCharacterKinematic_profile_common_cflags    += $(addprefix -I, $(PhysXCharacterKinematic_profile_hpaths))
 PhysXCharacterKinematic_profile_common_cflags  += -m64
-PhysXCharacterKinematic_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXCharacterKinematic_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXCharacterKinematic_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXCharacterKinematic_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXCharacterKinematic_profile_common_cflags  += -Wno-missing-field-initializers
 PhysXCharacterKinematic_profile_common_cflags  += -O3 -fno-strict-aliasing
+PhysXCharacterKinematic_profile_common_cflags  += $(DAVA_custom_flags)
 PhysXCharacterKinematic_profile_cflags	:= $(PhysXCharacterKinematic_profile_common_cflags)
 PhysXCharacterKinematic_profile_cppflags	:= $(PhysXCharacterKinematic_profile_common_cflags)
 PhysXCharacterKinematic_profile_lflags    := $(PhysXCharacterKinematic_custom_lflags)
@@ -398,11 +401,12 @@
 PhysXCharacterKinematic_release_common_cflags    += $(addprefix -D, $(PhysXCharacterKinematic_release_defines))
 PhysXCharacterKinematic_release_common_cflags    += $(addprefix -I, $(PhysXCharacterKinematic_release_hpaths))
 PhysXCharacterKinematic_release_common_cflags  += -m64
-PhysXCharacterKinematic_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXCharacterKinematic_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXCharacterKinematic_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXCharacterKinematic_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXCharacterKinematic_release_common_cflags  += -Wno-missing-field-initializers
 PhysXCharacterKinematic_release_common_cflags  += -O3 -fno-strict-aliasing
+PhysXCharacterKinematic_release_common_cflags  += $(DAVA_custom_flags)
 PhysXCharacterKinematic_release_cflags	:= $(PhysXCharacterKinematic_release_common_cflags)
 PhysXCharacterKinematic_release_cppflags	:= $(PhysXCharacterKinematic_release_common_cflags)
 PhysXCharacterKinematic_release_lflags    := $(PhysXCharacterKinematic_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXCommon.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXCommon.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXCommon.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXCommon.mk	2018-03-19 11:36:17.000000000 +0300
@@ -199,11 +199,12 @@
 PhysXCommon_release_common_cflags    += $(addprefix -D, $(PhysXCommon_release_defines))
 PhysXCommon_release_common_cflags    += $(addprefix -I, $(PhysXCommon_release_hpaths))
 PhysXCommon_release_common_cflags  += -m64
-PhysXCommon_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXCommon_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXCommon_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXCommon_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXCommon_release_common_cflags  += -Wno-missing-field-initializers
 PhysXCommon_release_common_cflags  += -O3 -fno-strict-aliasing
+PhysXCommon_release_common_cflags  += $(DAVA_custom_flags)
 PhysXCommon_release_cflags	:= $(PhysXCommon_release_common_cflags)
 PhysXCommon_release_cppflags	:= $(PhysXCommon_release_common_cflags)
 PhysXCommon_release_lflags    := $(PhysXCommon_custom_lflags)
@@ -309,11 +310,12 @@
 PhysXCommon_debug_common_cflags    += $(addprefix -D, $(PhysXCommon_debug_defines))
 PhysXCommon_debug_common_cflags    += $(addprefix -I, $(PhysXCommon_debug_hpaths))
 PhysXCommon_debug_common_cflags  += -m64
-PhysXCommon_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXCommon_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXCommon_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXCommon_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXCommon_debug_common_cflags  += -Wno-missing-field-initializers
 PhysXCommon_debug_common_cflags  += -g3 -gdwarf-2
+PhysXCommon_debug_common_cflags  += $(DAVA_custom_flags)
 PhysXCommon_debug_cflags	:= $(PhysXCommon_debug_common_cflags)
 PhysXCommon_debug_cppflags	:= $(PhysXCommon_debug_common_cflags)
 PhysXCommon_debug_lflags    := $(PhysXCommon_custom_lflags)
@@ -418,11 +420,12 @@
 PhysXCommon_checked_common_cflags    += $(addprefix -D, $(PhysXCommon_checked_defines))
 PhysXCommon_checked_common_cflags    += $(addprefix -I, $(PhysXCommon_checked_hpaths))
 PhysXCommon_checked_common_cflags  += -m64
-PhysXCommon_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXCommon_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXCommon_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXCommon_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXCommon_checked_common_cflags  += -Wno-missing-field-initializers
 PhysXCommon_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+PhysXCommon_checked_common_cflags  += $(DAVA_custom_flags)
 PhysXCommon_checked_cflags	:= $(PhysXCommon_checked_common_cflags)
 PhysXCommon_checked_cppflags	:= $(PhysXCommon_checked_common_cflags)
 PhysXCommon_checked_lflags    := $(PhysXCommon_custom_lflags)
@@ -527,11 +530,12 @@
 PhysXCommon_profile_common_cflags    += $(addprefix -D, $(PhysXCommon_profile_defines))
 PhysXCommon_profile_common_cflags    += $(addprefix -I, $(PhysXCommon_profile_hpaths))
 PhysXCommon_profile_common_cflags  += -m64
-PhysXCommon_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXCommon_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXCommon_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXCommon_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXCommon_profile_common_cflags  += -Wno-missing-field-initializers
 PhysXCommon_profile_common_cflags  += -O3 -fno-strict-aliasing
+PhysXCommon_profile_common_cflags  += $(DAVA_custom_flags)
 PhysXCommon_profile_cflags	:= $(PhysXCommon_profile_common_cflags)
 PhysXCommon_profile_cppflags	:= $(PhysXCommon_profile_common_cflags)
 PhysXCommon_profile_lflags    := $(PhysXCommon_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXCooking.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXCooking.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXCooking.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXCooking.mk	2018-03-19 11:36:17.000000000 +0300
@@ -92,11 +92,12 @@
 PhysXCooking_release_common_cflags    += $(addprefix -D, $(PhysXCooking_release_defines))
 PhysXCooking_release_common_cflags    += $(addprefix -I, $(PhysXCooking_release_hpaths))
 PhysXCooking_release_common_cflags  += -m64
-PhysXCooking_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXCooking_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXCooking_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXCooking_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXCooking_release_common_cflags  += -Wno-missing-field-initializers
 PhysXCooking_release_common_cflags  += -O3 -fno-strict-aliasing
+PhysXCooking_release_common_cflags  += $(DAVA_custom_flags)
 PhysXCooking_release_cflags	:= $(PhysXCooking_release_common_cflags)
 PhysXCooking_release_cppflags	:= $(PhysXCooking_release_common_cflags)
 PhysXCooking_release_lflags    := $(PhysXCooking_custom_lflags)
@@ -211,11 +212,12 @@
 PhysXCooking_debug_common_cflags    += $(addprefix -D, $(PhysXCooking_debug_defines))
 PhysXCooking_debug_common_cflags    += $(addprefix -I, $(PhysXCooking_debug_hpaths))
 PhysXCooking_debug_common_cflags  += -m64
-PhysXCooking_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXCooking_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXCooking_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXCooking_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXCooking_debug_common_cflags  += -Wno-missing-field-initializers
 PhysXCooking_debug_common_cflags  += -g3 -gdwarf-2
+PhysXCooking_debug_common_cflags  += $(DAVA_custom_flags)
 PhysXCooking_debug_cflags	:= $(PhysXCooking_debug_common_cflags)
 PhysXCooking_debug_cppflags	:= $(PhysXCooking_debug_common_cflags)
 PhysXCooking_debug_lflags    := $(PhysXCooking_custom_lflags)
@@ -329,11 +331,12 @@
 PhysXCooking_checked_common_cflags    += $(addprefix -D, $(PhysXCooking_checked_defines))
 PhysXCooking_checked_common_cflags    += $(addprefix -I, $(PhysXCooking_checked_hpaths))
 PhysXCooking_checked_common_cflags  += -m64
-PhysXCooking_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXCooking_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXCooking_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXCooking_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXCooking_checked_common_cflags  += -Wno-missing-field-initializers
 PhysXCooking_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+PhysXCooking_checked_common_cflags  += $(DAVA_custom_flags)
 PhysXCooking_checked_cflags	:= $(PhysXCooking_checked_common_cflags)
 PhysXCooking_checked_cppflags	:= $(PhysXCooking_checked_common_cflags)
 PhysXCooking_checked_lflags    := $(PhysXCooking_custom_lflags)
@@ -447,11 +450,12 @@
 PhysXCooking_profile_common_cflags    += $(addprefix -D, $(PhysXCooking_profile_defines))
 PhysXCooking_profile_common_cflags    += $(addprefix -I, $(PhysXCooking_profile_hpaths))
 PhysXCooking_profile_common_cflags  += -m64
-PhysXCooking_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXCooking_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXCooking_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXCooking_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXCooking_profile_common_cflags  += -Wno-missing-field-initializers
 PhysXCooking_profile_common_cflags  += -O3 -fno-strict-aliasing
+PhysXCooking_profile_common_cflags  += $(DAVA_custom_flags)
 PhysXCooking_profile_cflags	:= $(PhysXCooking_profile_common_cflags)
 PhysXCooking_profile_cppflags	:= $(PhysXCooking_profile_common_cflags)
 PhysXCooking_profile_lflags    := $(PhysXCooking_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXExtensions.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXExtensions.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXExtensions.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXExtensions.mk	2018-03-19 11:36:17.000000000 +0300
@@ -127,11 +127,12 @@
 PhysXExtensions_debug_common_cflags    += $(addprefix -D, $(PhysXExtensions_debug_defines))
 PhysXExtensions_debug_common_cflags    += $(addprefix -I, $(PhysXExtensions_debug_hpaths))
 PhysXExtensions_debug_common_cflags  += -m64
-PhysXExtensions_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXExtensions_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXExtensions_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXExtensions_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXExtensions_debug_common_cflags  += -Wno-missing-field-initializers
 PhysXExtensions_debug_common_cflags  += -g3 -gdwarf-2
+PhysXExtensions_debug_common_cflags  += $(DAVA_custom_flags)
 PhysXExtensions_debug_cflags	:= $(PhysXExtensions_debug_common_cflags)
 PhysXExtensions_debug_cppflags	:= $(PhysXExtensions_debug_common_cflags)
 PhysXExtensions_debug_lflags    := $(PhysXExtensions_custom_lflags)
@@ -243,11 +244,12 @@
 PhysXExtensions_checked_common_cflags    += $(addprefix -D, $(PhysXExtensions_checked_defines))
 PhysXExtensions_checked_common_cflags    += $(addprefix -I, $(PhysXExtensions_checked_hpaths))
 PhysXExtensions_checked_common_cflags  += -m64
-PhysXExtensions_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXExtensions_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXExtensions_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXExtensions_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXExtensions_checked_common_cflags  += -Wno-missing-field-initializers
 PhysXExtensions_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+PhysXExtensions_checked_common_cflags  += $(DAVA_custom_flags)
 PhysXExtensions_checked_cflags	:= $(PhysXExtensions_checked_common_cflags)
 PhysXExtensions_checked_cppflags	:= $(PhysXExtensions_checked_common_cflags)
 PhysXExtensions_checked_lflags    := $(PhysXExtensions_custom_lflags)
@@ -359,11 +361,12 @@
 PhysXExtensions_profile_common_cflags    += $(addprefix -D, $(PhysXExtensions_profile_defines))
 PhysXExtensions_profile_common_cflags    += $(addprefix -I, $(PhysXExtensions_profile_hpaths))
 PhysXExtensions_profile_common_cflags  += -m64
-PhysXExtensions_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXExtensions_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXExtensions_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXExtensions_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXExtensions_profile_common_cflags  += -Wno-missing-field-initializers
 PhysXExtensions_profile_common_cflags  += -O3 -fno-strict-aliasing
+PhysXExtensions_profile_common_cflags  += $(DAVA_custom_flags)
 PhysXExtensions_profile_cflags	:= $(PhysXExtensions_profile_common_cflags)
 PhysXExtensions_profile_cppflags	:= $(PhysXExtensions_profile_common_cflags)
 PhysXExtensions_profile_lflags    := $(PhysXExtensions_custom_lflags)
@@ -474,11 +477,12 @@
 PhysXExtensions_release_common_cflags    += $(addprefix -D, $(PhysXExtensions_release_defines))
 PhysXExtensions_release_common_cflags    += $(addprefix -I, $(PhysXExtensions_release_hpaths))
 PhysXExtensions_release_common_cflags  += -m64
-PhysXExtensions_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXExtensions_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXExtensions_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXExtensions_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXExtensions_release_common_cflags  += -Wno-missing-field-initializers
 PhysXExtensions_release_common_cflags  += -O3 -fno-strict-aliasing
+PhysXExtensions_release_common_cflags  += $(DAVA_custom_flags)
 PhysXExtensions_release_cflags	:= $(PhysXExtensions_release_common_cflags)
 PhysXExtensions_release_cppflags	:= $(PhysXExtensions_release_common_cflags)
 PhysXExtensions_release_lflags    := $(PhysXExtensions_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXVehicle.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXVehicle.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXVehicle.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PhysXVehicle.mk	2018-03-19 11:36:17.000000000 +0300
@@ -74,11 +74,12 @@
 PhysXVehicle_debug_common_cflags    += $(addprefix -D, $(PhysXVehicle_debug_defines))
 PhysXVehicle_debug_common_cflags    += $(addprefix -I, $(PhysXVehicle_debug_hpaths))
 PhysXVehicle_debug_common_cflags  += -m64
-PhysXVehicle_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXVehicle_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXVehicle_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXVehicle_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXVehicle_debug_common_cflags  += -Wno-missing-field-initializers
 PhysXVehicle_debug_common_cflags  += -g3 -gdwarf-2
+PhysXVehicle_debug_common_cflags  += $(DAVA_custom_flags)
 PhysXVehicle_debug_cflags	:= $(PhysXVehicle_debug_common_cflags)
 PhysXVehicle_debug_cppflags	:= $(PhysXVehicle_debug_common_cflags)
 PhysXVehicle_debug_lflags    := $(PhysXVehicle_custom_lflags)
@@ -173,11 +174,12 @@
 PhysXVehicle_checked_common_cflags    += $(addprefix -D, $(PhysXVehicle_checked_defines))
 PhysXVehicle_checked_common_cflags    += $(addprefix -I, $(PhysXVehicle_checked_hpaths))
 PhysXVehicle_checked_common_cflags  += -m64
-PhysXVehicle_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXVehicle_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXVehicle_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXVehicle_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXVehicle_checked_common_cflags  += -Wno-missing-field-initializers
 PhysXVehicle_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+PhysXVehicle_checked_common_cflags  += $(DAVA_custom_flags)
 PhysXVehicle_checked_cflags	:= $(PhysXVehicle_checked_common_cflags)
 PhysXVehicle_checked_cppflags	:= $(PhysXVehicle_checked_common_cflags)
 PhysXVehicle_checked_lflags    := $(PhysXVehicle_custom_lflags)
@@ -272,11 +274,12 @@
 PhysXVehicle_profile_common_cflags    += $(addprefix -D, $(PhysXVehicle_profile_defines))
 PhysXVehicle_profile_common_cflags    += $(addprefix -I, $(PhysXVehicle_profile_hpaths))
 PhysXVehicle_profile_common_cflags  += -m64
-PhysXVehicle_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXVehicle_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXVehicle_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXVehicle_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXVehicle_profile_common_cflags  += -Wno-missing-field-initializers
 PhysXVehicle_profile_common_cflags  += -O3 -fno-strict-aliasing
+PhysXVehicle_profile_common_cflags  += $(DAVA_custom_flags)
 PhysXVehicle_profile_cflags	:= $(PhysXVehicle_profile_common_cflags)
 PhysXVehicle_profile_cppflags	:= $(PhysXVehicle_profile_common_cflags)
 PhysXVehicle_profile_lflags    := $(PhysXVehicle_custom_lflags)
@@ -370,11 +373,12 @@
 PhysXVehicle_release_common_cflags    += $(addprefix -D, $(PhysXVehicle_release_defines))
 PhysXVehicle_release_common_cflags    += $(addprefix -I, $(PhysXVehicle_release_hpaths))
 PhysXVehicle_release_common_cflags  += -m64
-PhysXVehicle_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PhysXVehicle_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PhysXVehicle_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PhysXVehicle_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PhysXVehicle_release_common_cflags  += -Wno-missing-field-initializers
 PhysXVehicle_release_common_cflags  += -O3 -fno-strict-aliasing
+PhysXVehicle_release_common_cflags  += $(DAVA_custom_flags)
 PhysXVehicle_release_cflags	:= $(PhysXVehicle_release_common_cflags)
 PhysXVehicle_release_cppflags	:= $(PhysXVehicle_release_common_cflags)
 PhysXVehicle_release_lflags    := $(PhysXVehicle_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PsFastXml.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PsFastXml.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PsFastXml.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PsFastXml.mk	2018-03-19 11:36:17.000000000 +0300
@@ -39,11 +39,12 @@
 PsFastXml_debug_common_cflags    += $(addprefix -D, $(PsFastXml_debug_defines))
 PsFastXml_debug_common_cflags    += $(addprefix -I, $(PsFastXml_debug_hpaths))
 PsFastXml_debug_common_cflags  += -m64
-PsFastXml_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PsFastXml_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PsFastXml_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PsFastXml_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PsFastXml_debug_common_cflags  += -Wno-missing-field-initializers
 PsFastXml_debug_common_cflags  += -g3 -gdwarf-2
+PsFastXml_debug_common_cflags += $(DAVA_custom_flags)
 PsFastXml_debug_cflags	:= $(PsFastXml_debug_common_cflags)
 PsFastXml_debug_cppflags	:= $(PsFastXml_debug_common_cflags)
 PsFastXml_debug_lflags    := $(PsFastXml_custom_lflags)
@@ -119,11 +120,12 @@
 PsFastXml_release_common_cflags    += $(addprefix -D, $(PsFastXml_release_defines))
 PsFastXml_release_common_cflags    += $(addprefix -I, $(PsFastXml_release_hpaths))
 PsFastXml_release_common_cflags  += -m64
-PsFastXml_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PsFastXml_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PsFastXml_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PsFastXml_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PsFastXml_release_common_cflags  += -Wno-missing-field-initializers
 PsFastXml_release_common_cflags  += -O3 -fno-strict-aliasing
+PsFastXml_release_common_cflags  += $(DAVA_custom_flags)
 PsFastXml_release_cflags	:= $(PsFastXml_release_common_cflags)
 PsFastXml_release_cppflags	:= $(PsFastXml_release_common_cflags)
 PsFastXml_release_lflags    := $(PsFastXml_custom_lflags)
@@ -200,11 +202,12 @@
 PsFastXml_checked_common_cflags    += $(addprefix -D, $(PsFastXml_checked_defines))
 PsFastXml_checked_common_cflags    += $(addprefix -I, $(PsFastXml_checked_hpaths))
 PsFastXml_checked_common_cflags  += -m64
-PsFastXml_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PsFastXml_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PsFastXml_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PsFastXml_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PsFastXml_checked_common_cflags  += -Wno-missing-field-initializers
 PsFastXml_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+PsFastXml_checked_common_cflags  += $(DAVA_custom_flags)
 PsFastXml_checked_cflags	:= $(PsFastXml_checked_common_cflags)
 PsFastXml_checked_cppflags	:= $(PsFastXml_checked_common_cflags)
 PsFastXml_checked_lflags    := $(PsFastXml_custom_lflags)
@@ -281,11 +284,12 @@
 PsFastXml_profile_common_cflags    += $(addprefix -D, $(PsFastXml_profile_defines))
 PsFastXml_profile_common_cflags    += $(addprefix -I, $(PsFastXml_profile_hpaths))
 PsFastXml_profile_common_cflags  += -m64
-PsFastXml_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PsFastXml_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PsFastXml_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PsFastXml_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PsFastXml_profile_common_cflags  += -Wno-missing-field-initializers
 PsFastXml_profile_common_cflags  += -O3 -fno-strict-aliasing
+PsFastXml_profile_common_cflags  += $(DAVA_custom_flags)
 PsFastXml_profile_cflags	:= $(PsFastXml_profile_common_cflags)
 PsFastXml_profile_cppflags	:= $(PsFastXml_profile_common_cflags)
 PsFastXml_profile_lflags    := $(PsFastXml_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PxFoundation.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PxFoundation.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PxFoundation.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PxFoundation.mk	2018-03-19 11:36:17.000000000 +0300
@@ -55,11 +55,12 @@
 PxFoundation_debug_common_cflags    += $(addprefix -D, $(PxFoundation_debug_defines))
 PxFoundation_debug_common_cflags    += $(addprefix -I, $(PxFoundation_debug_hpaths))
 PxFoundation_debug_common_cflags  += -m64
-PxFoundation_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PxFoundation_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PxFoundation_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PxFoundation_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PxFoundation_debug_common_cflags  += -Wno-missing-field-initializers
 PxFoundation_debug_common_cflags  += -g3 -gdwarf-2
+PxFoundation_debug_common_cflags  += $(DAVA_custom_flags)
 PxFoundation_debug_cflags	:= $(PxFoundation_debug_common_cflags)
 PxFoundation_debug_cppflags	:= $(PxFoundation_debug_common_cflags)
 PxFoundation_debug_lflags    := $(PxFoundation_custom_lflags)
@@ -135,11 +136,12 @@
 PxFoundation_release_common_cflags    += $(addprefix -D, $(PxFoundation_release_defines))
 PxFoundation_release_common_cflags    += $(addprefix -I, $(PxFoundation_release_hpaths))
 PxFoundation_release_common_cflags  += -m64
-PxFoundation_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PxFoundation_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PxFoundation_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PxFoundation_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PxFoundation_release_common_cflags  += -Wno-missing-field-initializers
 PxFoundation_release_common_cflags  += -O3 -fno-strict-aliasing
+PxFoundation_release_common_cflags  += $(DAVA_custom_flags)
 PxFoundation_release_cflags	:= $(PxFoundation_release_common_cflags)
 PxFoundation_release_cppflags	:= $(PxFoundation_release_common_cflags)
 PxFoundation_release_lflags    := $(PxFoundation_custom_lflags)
@@ -216,11 +218,12 @@
 PxFoundation_checked_common_cflags    += $(addprefix -D, $(PxFoundation_checked_defines))
 PxFoundation_checked_common_cflags    += $(addprefix -I, $(PxFoundation_checked_hpaths))
 PxFoundation_checked_common_cflags  += -m64
-PxFoundation_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PxFoundation_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PxFoundation_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PxFoundation_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PxFoundation_checked_common_cflags  += -Wno-missing-field-initializers
 PxFoundation_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+PxFoundation_checked_common_cflags  += $(DAVA_custom_flags)
 PxFoundation_checked_cflags	:= $(PxFoundation_checked_common_cflags)
 PxFoundation_checked_cppflags	:= $(PxFoundation_checked_common_cflags)
 PxFoundation_checked_lflags    := $(PxFoundation_custom_lflags)
@@ -297,11 +300,12 @@
 PxFoundation_profile_common_cflags    += $(addprefix -D, $(PxFoundation_profile_defines))
 PxFoundation_profile_common_cflags    += $(addprefix -I, $(PxFoundation_profile_hpaths))
 PxFoundation_profile_common_cflags  += -m64
-PxFoundation_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PxFoundation_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PxFoundation_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PxFoundation_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PxFoundation_profile_common_cflags  += -Wno-missing-field-initializers
 PxFoundation_profile_common_cflags  += -O3 -fno-strict-aliasing
+PxFoundation_profile_common_cflags  += $(DAVA_custom_flags)
 PxFoundation_profile_cflags	:= $(PxFoundation_profile_common_cflags)
 PxFoundation_profile_cppflags	:= $(PxFoundation_profile_common_cflags)
 PxFoundation_profile_lflags    := $(PxFoundation_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PxPvdSDK.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PxPvdSDK.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PxPvdSDK.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PxPvdSDK.mk	2018-03-19 11:36:17.000000000 +0300
@@ -54,11 +54,12 @@
 PxPvdSDK_debug_common_cflags    += $(addprefix -D, $(PxPvdSDK_debug_defines))
 PxPvdSDK_debug_common_cflags    += $(addprefix -I, $(PxPvdSDK_debug_hpaths))
 PxPvdSDK_debug_common_cflags  += -m64
-PxPvdSDK_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PxPvdSDK_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PxPvdSDK_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PxPvdSDK_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PxPvdSDK_debug_common_cflags  += -Wno-missing-field-initializers
 PxPvdSDK_debug_common_cflags  += -g3 -gdwarf-2
+PxPvdSDK_debug_common_cflags  += $(DAVA_custom_flags)
 PxPvdSDK_debug_cflags	:= $(PxPvdSDK_debug_common_cflags)
 PxPvdSDK_debug_cppflags	:= $(PxPvdSDK_debug_common_cflags)
 PxPvdSDK_debug_lflags    := $(PxPvdSDK_custom_lflags)
@@ -139,11 +140,12 @@
 PxPvdSDK_release_common_cflags    += $(addprefix -D, $(PxPvdSDK_release_defines))
 PxPvdSDK_release_common_cflags    += $(addprefix -I, $(PxPvdSDK_release_hpaths))
 PxPvdSDK_release_common_cflags  += -m64
-PxPvdSDK_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PxPvdSDK_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PxPvdSDK_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PxPvdSDK_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PxPvdSDK_release_common_cflags  += -Wno-missing-field-initializers
 PxPvdSDK_release_common_cflags  += -O3 -fno-strict-aliasing
+PxPvdSDK_release_common_cflags  += $(DAVA_custom_flags)
 PxPvdSDK_release_cflags	:= $(PxPvdSDK_release_common_cflags)
 PxPvdSDK_release_cppflags	:= $(PxPvdSDK_release_common_cflags)
 PxPvdSDK_release_lflags    := $(PxPvdSDK_custom_lflags)
@@ -225,11 +227,12 @@
 PxPvdSDK_checked_common_cflags    += $(addprefix -D, $(PxPvdSDK_checked_defines))
 PxPvdSDK_checked_common_cflags    += $(addprefix -I, $(PxPvdSDK_checked_hpaths))
 PxPvdSDK_checked_common_cflags  += -m64
-PxPvdSDK_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PxPvdSDK_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PxPvdSDK_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PxPvdSDK_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PxPvdSDK_checked_common_cflags  += -Wno-missing-field-initializers
 PxPvdSDK_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+PxPvdSDK_checked_common_cflags  += $(DAVA_custom_flags)
 PxPvdSDK_checked_cflags	:= $(PxPvdSDK_checked_common_cflags)
 PxPvdSDK_checked_cppflags	:= $(PxPvdSDK_checked_common_cflags)
 PxPvdSDK_checked_lflags    := $(PxPvdSDK_custom_lflags)
@@ -311,11 +314,12 @@
 PxPvdSDK_profile_common_cflags    += $(addprefix -D, $(PxPvdSDK_profile_defines))
 PxPvdSDK_profile_common_cflags    += $(addprefix -I, $(PxPvdSDK_profile_hpaths))
 PxPvdSDK_profile_common_cflags  += -m64
-PxPvdSDK_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PxPvdSDK_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PxPvdSDK_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PxPvdSDK_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PxPvdSDK_profile_common_cflags  += -Wno-missing-field-initializers
 PxPvdSDK_profile_common_cflags  += -O3 -fno-strict-aliasing
+PxPvdSDK_profile_common_cflags  += $(DAVA_custom_flags)
 PxPvdSDK_profile_cflags	:= $(PxPvdSDK_profile_common_cflags)
 PxPvdSDK_profile_cppflags	:= $(PxPvdSDK_profile_common_cflags)
 PxPvdSDK_profile_lflags    := $(PxPvdSDK_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PxTask.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PxTask.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.PxTask.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.PxTask.mk	2018-03-19 11:36:17.000000000 +0300
@@ -38,11 +38,12 @@
 PxTask_debug_common_cflags    += $(addprefix -D, $(PxTask_debug_defines))
 PxTask_debug_common_cflags    += $(addprefix -I, $(PxTask_debug_hpaths))
 PxTask_debug_common_cflags  += -m64
-PxTask_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PxTask_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PxTask_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PxTask_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PxTask_debug_common_cflags  += -Wno-missing-field-initializers
 PxTask_debug_common_cflags  += -g3 -gdwarf-2
+PxTask_debug_common_cflags  += $(DAVA_custom_flags)
 PxTask_debug_cflags	:= $(PxTask_debug_common_cflags)
 PxTask_debug_cppflags	:= $(PxTask_debug_common_cflags)
 PxTask_debug_lflags    := $(PxTask_custom_lflags)
@@ -117,11 +118,12 @@
 PxTask_release_common_cflags    += $(addprefix -D, $(PxTask_release_defines))
 PxTask_release_common_cflags    += $(addprefix -I, $(PxTask_release_hpaths))
 PxTask_release_common_cflags  += -m64
-PxTask_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PxTask_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PxTask_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PxTask_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PxTask_release_common_cflags  += -Wno-missing-field-initializers
 PxTask_release_common_cflags  += -O3 -fno-strict-aliasing
+PxTask_release_common_cflags  += $(DAVA_custom_flags)
 PxTask_release_cflags	:= $(PxTask_release_common_cflags)
 PxTask_release_cppflags	:= $(PxTask_release_common_cflags)
 PxTask_release_lflags    := $(PxTask_custom_lflags)
@@ -197,11 +199,12 @@
 PxTask_checked_common_cflags    += $(addprefix -D, $(PxTask_checked_defines))
 PxTask_checked_common_cflags    += $(addprefix -I, $(PxTask_checked_hpaths))
 PxTask_checked_common_cflags  += -m64
-PxTask_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PxTask_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PxTask_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PxTask_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PxTask_checked_common_cflags  += -Wno-missing-field-initializers
 PxTask_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+PxTask_checked_common_cflags  += $(DAVA_custom_flags)
 PxTask_checked_cflags	:= $(PxTask_checked_common_cflags)
 PxTask_checked_cppflags	:= $(PxTask_checked_common_cflags)
 PxTask_checked_lflags    := $(PxTask_custom_lflags)
@@ -277,11 +280,12 @@
 PxTask_profile_common_cflags    += $(addprefix -D, $(PxTask_profile_defines))
 PxTask_profile_common_cflags    += $(addprefix -I, $(PxTask_profile_hpaths))
 PxTask_profile_common_cflags  += -m64
-PxTask_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+PxTask_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 PxTask_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 PxTask_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 PxTask_profile_common_cflags  += -Wno-missing-field-initializers
 PxTask_profile_common_cflags  += -O3 -fno-strict-aliasing
+PxTask_profile_common_cflags  += $(DAVA_custom_flags)
 PxTask_profile_cflags	:= $(PxTask_profile_common_cflags)
 PxTask_profile_cppflags	:= $(PxTask_profile_common_cflags)
 PxTask_profile_lflags    := $(PxTask_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.SceneQuery.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.SceneQuery.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.SceneQuery.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.SceneQuery.mk	2018-03-19 11:36:17.000000000 +0300
@@ -76,11 +76,12 @@
 SceneQuery_debug_common_cflags    += $(addprefix -D, $(SceneQuery_debug_defines))
 SceneQuery_debug_common_cflags    += $(addprefix -I, $(SceneQuery_debug_hpaths))
 SceneQuery_debug_common_cflags  += -m64
-SceneQuery_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+SceneQuery_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 SceneQuery_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 SceneQuery_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 SceneQuery_debug_common_cflags  += -Wno-missing-field-initializers
 SceneQuery_debug_common_cflags  += -g3 -gdwarf-2
+SceneQuery_debug_common_cflags  += $(DAVA_custom_flags)
 SceneQuery_debug_cflags	:= $(SceneQuery_debug_common_cflags)
 SceneQuery_debug_cppflags	:= $(SceneQuery_debug_common_cflags)
 SceneQuery_debug_lflags    := $(SceneQuery_custom_lflags)
@@ -185,11 +186,12 @@
 SceneQuery_checked_common_cflags    += $(addprefix -D, $(SceneQuery_checked_defines))
 SceneQuery_checked_common_cflags    += $(addprefix -I, $(SceneQuery_checked_hpaths))
 SceneQuery_checked_common_cflags  += -m64
-SceneQuery_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+SceneQuery_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 SceneQuery_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 SceneQuery_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 SceneQuery_checked_common_cflags  += -Wno-missing-field-initializers
 SceneQuery_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+SceneQuery_checked_common_cflags  += $(DAVA_custom_flags)
 SceneQuery_checked_cflags	:= $(SceneQuery_checked_common_cflags)
 SceneQuery_checked_cppflags	:= $(SceneQuery_checked_common_cflags)
 SceneQuery_checked_lflags    := $(SceneQuery_custom_lflags)
@@ -294,11 +296,12 @@
 SceneQuery_profile_common_cflags    += $(addprefix -D, $(SceneQuery_profile_defines))
 SceneQuery_profile_common_cflags    += $(addprefix -I, $(SceneQuery_profile_hpaths))
 SceneQuery_profile_common_cflags  += -m64
-SceneQuery_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+SceneQuery_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 SceneQuery_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 SceneQuery_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 SceneQuery_profile_common_cflags  += -Wno-missing-field-initializers
 SceneQuery_profile_common_cflags  += -O3 -fno-strict-aliasing
+SceneQuery_profile_common_cflags  += $(DAVA_custom_flags)
 SceneQuery_profile_cflags	:= $(SceneQuery_profile_common_cflags)
 SceneQuery_profile_cppflags	:= $(SceneQuery_profile_common_cflags)
 SceneQuery_profile_lflags    := $(SceneQuery_custom_lflags)
@@ -402,11 +405,12 @@
 SceneQuery_release_common_cflags    += $(addprefix -D, $(SceneQuery_release_defines))
 SceneQuery_release_common_cflags    += $(addprefix -I, $(SceneQuery_release_hpaths))
 SceneQuery_release_common_cflags  += -m64
-SceneQuery_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+SceneQuery_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 SceneQuery_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 SceneQuery_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 SceneQuery_release_common_cflags  += -Wno-missing-field-initializers
 SceneQuery_release_common_cflags  += -O3 -fno-strict-aliasing
+SceneQuery_release_common_cflags  += $(DAVA_custom_flags)
 SceneQuery_release_cflags	:= $(SceneQuery_release_common_cflags)
 SceneQuery_release_cppflags	:= $(SceneQuery_release_common_cflags)
 SceneQuery_release_lflags    := $(SceneQuery_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.SimulationController.mk ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.SimulationController.mk
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.SimulationController.mk	2017-04-25 14:02:08.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.SimulationController.mk	2018-03-19 11:36:17.000000000 +0300
@@ -122,11 +122,12 @@
 SimulationController_debug_common_cflags    += $(addprefix -D, $(SimulationController_debug_defines))
 SimulationController_debug_common_cflags    += $(addprefix -I, $(SimulationController_debug_hpaths))
 SimulationController_debug_common_cflags  += -m64
-SimulationController_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+SimulationController_debug_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 SimulationController_debug_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 SimulationController_debug_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 SimulationController_debug_common_cflags  += -Wno-missing-field-initializers
 SimulationController_debug_common_cflags  += -g3 -gdwarf-2
+SimulationController_debug_common_cflags  += $(DAVA_custom_flags)
 SimulationController_debug_cflags	:= $(SimulationController_debug_common_cflags)
 SimulationController_debug_cppflags	:= $(SimulationController_debug_common_cflags)
 SimulationController_debug_lflags    := $(SimulationController_custom_lflags)
@@ -244,11 +245,12 @@
 SimulationController_checked_common_cflags    += $(addprefix -D, $(SimulationController_checked_defines))
 SimulationController_checked_common_cflags    += $(addprefix -I, $(SimulationController_checked_hpaths))
 SimulationController_checked_common_cflags  += -m64
-SimulationController_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+SimulationController_checked_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 SimulationController_checked_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 SimulationController_checked_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 SimulationController_checked_common_cflags  += -Wno-missing-field-initializers
 SimulationController_checked_common_cflags  += -g3 -gdwarf-2 -O3 -fno-strict-aliasing
+SimulationController_checked_common_cflags  += $(DAVA_custom_flags)
 SimulationController_checked_cflags	:= $(SimulationController_checked_common_cflags)
 SimulationController_checked_cppflags	:= $(SimulationController_checked_common_cflags)
 SimulationController_checked_lflags    := $(SimulationController_custom_lflags)
@@ -366,11 +368,12 @@
 SimulationController_profile_common_cflags    += $(addprefix -D, $(SimulationController_profile_defines))
 SimulationController_profile_common_cflags    += $(addprefix -I, $(SimulationController_profile_hpaths))
 SimulationController_profile_common_cflags  += -m64
-SimulationController_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+SimulationController_profile_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 SimulationController_profile_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 SimulationController_profile_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 SimulationController_profile_common_cflags  += -Wno-missing-field-initializers
 SimulationController_profile_common_cflags  += -O3 -fno-strict-aliasing
+SimulationController_profile_common_cflags  += $(DAVA_custom_flags)
 SimulationController_profile_cflags	:= $(SimulationController_profile_common_cflags)
 SimulationController_profile_cppflags	:= $(SimulationController_profile_common_cflags)
 SimulationController_profile_lflags    := $(SimulationController_custom_lflags)
@@ -487,11 +490,12 @@
 SimulationController_release_common_cflags    += $(addprefix -D, $(SimulationController_release_defines))
 SimulationController_release_common_cflags    += $(addprefix -I, $(SimulationController_release_hpaths))
 SimulationController_release_common_cflags  += -m64
-SimulationController_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -ffast-math -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
+SimulationController_release_common_cflags  += -Werror -m64 -fPIC -msse2 -mfpmath=sse -fno-exceptions -fno-rtti -fvisibility=hidden -fvisibility-inlines-hidden
 SimulationController_release_common_cflags  += -Wall -Wextra -Wstrict-aliasing=2 -fdiagnostics-show-option
 SimulationController_release_common_cflags  += -Wno-invalid-offsetof -Wno-uninitialized
 SimulationController_release_common_cflags  += -Wno-missing-field-initializers
 SimulationController_release_common_cflags  += -O3 -fno-strict-aliasing
+SimulationController_release_common_cflags  += $(DAVA_custom_flags)
 SimulationController_release_cflags	:= $(SimulationController_release_common_cflags)
 SimulationController_release_cppflags	:= $(SimulationController_release_common_cflags)
 SimulationController_release_lflags    := $(SimulationController_custom_lflags)
diff -ur --new-file ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.custom ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.custom
--- ./PhysX-3.4-master_original/PhysX_3.4/Source/compiler/linux64/Makefile.custom	1970-01-01 03:00:00.000000000 +0300
+++ ./PhysX-3.4-master/PhysX_3.4/Source/compiler/linux64/Makefile.custom	2018-03-19 11:25:20.000000000 +0300
@@ -0,0 +1 @@
+DAVA_custom_flags = -Wno-return-type-c-linkage -Wno-invalid-noreturn -Wno-unused-local-typedef -Wno-undefined-bool-conversion -Wno-unused-private-field
