diff -ur ./freetype_source_original/CMakeLists.txt ./freetype_source/CMakeLists.txt
--- ./freetype_source_original/CMakeLists.txt	2016-09-08 09:34:15.000000000 +0300
+++ ./freetype_source/CMakeLists.txt	2017-08-18 10:13:38.368690200 +0300
@@ -118,6 +118,20 @@
 
 project(freetype)
 
+if (MSVC AND NOT (CMAKE_SYSTEM_NAME STREQUAL "WindowsStore"))
+    # setup static linking with crt
+    set(CompilerFlags
+            CMAKE_CXX_FLAGS
+            CMAKE_CXX_FLAGS_DEBUG
+            CMAKE_CXX_FLAGS_RELEASE
+            CMAKE_C_FLAGS
+            CMAKE_C_FLAGS_DEBUG
+            CMAKE_C_FLAGS_RELEASE
+    )
+    foreach(CompilerFlag ${CompilerFlags})
+        string(REPLACE "/MD" "/MT" ${CompilerFlag} "${${CompilerFlag}}")
+    endforeach()
+endif()
 
 if (WIN32 AND NOT MINGW AND BUILD_SHARED_LIBS)
   message(FATAL_ERROR "Building shared libraries on Windows needs MinGW")
