Only in ./libpsd_source/: CMakeLists.txt
diff -ur ./libpsd_source_original/include/libpsd.h ./libpsd_source/include/libpsd.h
--- ./libpsd_source_original/include/libpsd.h	2007-02-06 19:42:32.000000000 +0300
+++ ./libpsd_source/include/libpsd.h	2017-08-19 14:33:52.696534400 +0300
@@ -255,7 +255,7 @@
 	psd_layer_effects_bevel_emboss_inner_shadow,
 	psd_layer_effects_bevel_emboss_inner_light,
 	psd_layer_effects_bevel_emboss_texture
-};
+} psd_layer_effects_bevel_emboss_type;
 #define psd_layer_effects_image_count		(psd_layer_effects_bevel_emboss_texture + 1)
 
 typedef enum {
@@ -614,7 +614,7 @@
 	
 	psd_int						layer_info_count;
 	psd_layer_info_type			layer_info_type[psd_layer_info_type_count];
-	psd_uint					layer_info_data[psd_layer_info_type_count];
+	psd_pointer					layer_info_data[psd_layer_info_type_count];
 	psd_bool					adjustment_valid;
 	
 	psd_uchar 					fill_opacity;
diff -ur ./libpsd_source_original/include/psd_types.h ./libpsd_source/include/psd_types.h
--- ./libpsd_source_original/include/psd_types.h	2007-02-01 14:18:00.000000000 +0300
+++ ./libpsd_source/include/psd_types.h	2017-08-19 14:29:46.952639100 +0300
@@ -2,6 +2,7 @@
 #define __PSD_TYPES_H__
 
 #include <stddef.h>
+#include <stdint.h>
 
 
 typedef unsigned char			psd_bool;
@@ -17,7 +18,7 @@
 typedef unsigned int			psd_uint;
 typedef float					psd_float;
 typedef double					psd_double;
-
+typedef uintptr_t				psd_pointer;
 
 typedef unsigned char			psd_color_component;
 typedef unsigned int			psd_argb_color;
diff -ur ./libpsd_source_original/src/adjustment.c ./libpsd_source/src/adjustment.c
--- ./libpsd_source_original/src/adjustment.c	2007-01-28 19:40:58.000000000 +0300
+++ ./libpsd_source/src/adjustment.c	2017-08-19 14:37:51.261755600 +0300
@@ -33,7 +33,7 @@
 #include "psd_math.h"
 
 
-typedef void psd_adjustment_blend_proc(psd_uint layer_info_data, psd_int * red, psd_int * green, psd_int * blue);
+typedef void psd_adjustment_blend_proc(psd_pointer layer_info_data, psd_int * red, psd_int * green, psd_int * blue);
 
 extern psd_bool psd_layer_blend_levels(psd_context * context, psd_layer_record * layer, psd_rect * dst_rect);
 extern psd_bool psd_layer_blend_curves(psd_context * context, psd_layer_record * layer, psd_rect * dst_rect);
@@ -231,7 +231,7 @@
 }
 
 void psd_adjustment_blend_color(psd_context * context, psd_layer_record * layer, psd_rect * dst_rect, 
-	psd_adjustment_blend_proc * blend_proc, psd_uint layer_info_data)
+	psd_adjustment_blend_proc * blend_proc, psd_pointer layer_info_data)
 {
 	psd_int i, j, width, height;
 	psd_int dst_red, dst_green, dst_blue, dst_alpha, src_red, src_green, src_blue;
diff -ur ./libpsd_source_original/src/bevel_emboss.c ./libpsd_source/src/bevel_emboss.c
--- ./libpsd_source_original/src/bevel_emboss.c	2007-02-01 14:08:12.000000000 +0300
+++ ./libpsd_source/src/bevel_emboss.c	2017-08-19 14:22:05.640661600 +0300
@@ -24,6 +24,7 @@
  */
 
 #include <math.h>
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
diff -ur ./libpsd_source_original/src/brightness_contrast.c ./libpsd_source/src/brightness_contrast.c
--- ./libpsd_source_original/src/brightness_contrast.c	2007-01-28 19:32:42.000000000 +0300
+++ ./libpsd_source/src/brightness_contrast.c	2017-08-19 14:39:52.816532500 +0300
@@ -24,6 +24,7 @@
  */
 
 #include <math.h>
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -47,7 +48,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_brightness_contrast));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 
 	// Brightness
diff -ur ./libpsd_source_original/src/channel_image.c ./libpsd_source/src/channel_image.c
--- ./libpsd_source_original/src/channel_image.c	2007-02-03 13:14:54.000000000 +0300
+++ ./libpsd_source/src/channel_image.c	2017-08-19 14:25:15.987519100 +0300
@@ -23,6 +23,7 @@
  * $Id: channel_image.c, created by Patrick in 2006.05.25, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/channel_mixer.c ./libpsd_source/src/channel_mixer.c
--- ./libpsd_source_original/src/channel_mixer.c	2007-01-28 19:32:02.000000000 +0300
+++ ./libpsd_source/src/channel_mixer.c	2017-08-19 14:39:52.810527000 +0300
@@ -24,6 +24,7 @@
  */
 
 #include <math.h>
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -32,9 +33,9 @@
 #include "psd_math.h"
 
 
-typedef void psd_adjustment_blend_proc(psd_uint layer_info_data, psd_int * red, psd_int * green, psd_int * blue);
+typedef void psd_adjustment_blend_proc(psd_pointer layer_info_data, psd_int * red, psd_int * green, psd_int * blue);
 extern void psd_adjustment_blend_color(psd_context * context, psd_layer_record * layer, psd_rect * dst_rect, 
-	psd_adjustment_blend_proc * blend_proc, psd_uint layer_info_data);
+	psd_adjustment_blend_proc * blend_proc, psd_pointer layer_info_data);
 
 
 psd_status psd_get_layer_channel_mixer(psd_context * context, psd_layer_record * layer)
@@ -49,7 +50,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_channel_mixer));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 
 	// Version ( = 1)
@@ -75,7 +76,7 @@
 	return psd_status_done;
 }
 
-static void psd_channel_mixer_proc(psd_uint layer_info_data, psd_int * red, psd_int * green, psd_int * blue)
+static void psd_channel_mixer_proc(psd_pointer layer_info_data, psd_int * red, psd_int * green, psd_int * blue)
 {
 	psd_layer_channel_mixer * data = (psd_layer_channel_mixer *)layer_info_data;
 	psd_int dst_red, dst_green, dst_blue, src_red, src_green, src_blue, gray;
@@ -135,7 +136,7 @@
 
 psd_bool psd_layer_blend_channel_mixer(psd_context * context, psd_layer_record * layer, psd_rect * dst_rect)
 {
-	psd_uint layer_info_data;
+    psd_pointer layer_info_data;
 	psd_int i;
 
 	if(context->color_mode != psd_color_mode_rgb)
diff -ur ./libpsd_source_original/src/color.c ./libpsd_source/src/color.c
--- ./libpsd_source_original/src/color.c	2007-02-01 14:10:04.000000000 +0300
+++ ./libpsd_source/src/color.c	2017-08-19 14:25:15.974505700 +0300
@@ -24,6 +24,7 @@
  */
 
 #include <math.h>
+#include <string.h>
 #include "libpsd.h"
 #include "psd_color.h"
 #include "psd_math.h"
diff -ur ./libpsd_source_original/src/color_balance.c ./libpsd_source/src/color_balance.c
--- ./libpsd_source_original/src/color_balance.c	2007-01-28 19:29:24.000000000 +0300
+++ ./libpsd_source/src/color_balance.c	2017-08-19 14:39:52.804521100 +0300
@@ -24,6 +24,7 @@
  */
 
 #include <math.h>
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -63,7 +64,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_color_balance));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 
 	// shadows, midtones, highlights
diff -ur ./libpsd_source_original/src/color_overlay.c ./libpsd_source/src/color_overlay.c
--- ./libpsd_source_original/src/color_overlay.c	2007-01-30 16:45:38.000000000 +0300
+++ ./libpsd_source/src/color_overlay.c	2017-08-19 14:25:15.961493100 +0300
@@ -23,6 +23,7 @@
  * $Id: color_overlay.c, created by Patrick in 2006.05.23, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/curves.c ./libpsd_source/src/curves.c
--- ./libpsd_source_original/src/curves.c	2007-02-01 14:10:10.000000000 +0300
+++ ./libpsd_source/src/curves.c	2017-08-19 14:39:52.797513900 +0300
@@ -24,6 +24,7 @@
  */
 
 #include <math.h>
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -65,7 +66,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_curves));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 
 	// padding, document is wrong, maybe photoshop is wrong
@@ -175,7 +176,7 @@
 	return psd_status_done;
 }
 
-void psd_layer_curves_free(psd_uint info_data)
+void psd_layer_curves_free(psd_pointer info_data)
 {
 	psd_layer_curves * data;
 
diff -ur ./libpsd_source_original/src/descriptor.c ./libpsd_source/src/descriptor.c
--- ./libpsd_source_original/src/descriptor.c	2007-02-01 14:10:22.000000000 +0300
+++ ./libpsd_source/src/descriptor.c	2017-08-19 14:25:15.949482700 +0300
@@ -24,6 +24,7 @@
  */
 
 #include <math.h>
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_descriptor.h"
diff -ur ./libpsd_source_original/src/drop_shadow.c ./libpsd_source/src/drop_shadow.c
--- ./libpsd_source_original/src/drop_shadow.c	2007-02-01 14:10:26.000000000 +0300
+++ ./libpsd_source/src/drop_shadow.c	2017-08-19 14:25:15.943476200 +0300
@@ -24,6 +24,7 @@
  */
 
 #include <math.h>
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/effects.c ./libpsd_source/src/effects.c
--- ./libpsd_source_original/src/effects.c	2007-01-30 16:46:22.000000000 +0300
+++ ./libpsd_source/src/effects.c	2017-08-19 14:35:41.208099500 +0300
@@ -23,6 +23,7 @@
  * $Id: effects.c, created by Patrick in 2006.05.23, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
@@ -80,7 +81,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_effects));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)(void *)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)(void *)data;
 	layer->layer_info_count ++;
 
 	// Version: 0
@@ -170,7 +171,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_effects));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)(void *)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)(void *)data;
 	layer->layer_info_count ++;
 	
 	// Object effects version: 0
@@ -782,7 +783,7 @@
 	psd_freeif(pattern_info->name);
 }
 
-void psd_layer_effects_free(psd_uint layer_info)
+void psd_layer_effects_free(psd_pointer layer_info)
 {
 	psd_layer_effects * data;
 	psd_int i;
diff -ur ./libpsd_source_original/src/file_header.c ./libpsd_source/src/file_header.c
--- ./libpsd_source_original/src/file_header.c	2007-02-04 10:53:48.000000000 +0300
+++ ./libpsd_source/src/file_header.c	2017-08-19 14:25:15.930463900 +0300
@@ -23,6 +23,7 @@
  * $Id: file_header.c, created by Patrick in 2006.05.18, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_color.h"
diff -ur ./libpsd_source_original/src/gradient_blend.c ./libpsd_source/src/gradient_blend.c
--- ./libpsd_source_original/src/gradient_blend.c	2007-01-28 19:21:46.000000000 +0300
+++ ./libpsd_source/src/gradient_blend.c	2017-08-19 14:25:15.923457300 +0300
@@ -23,6 +23,7 @@
  * $Id: gradient_blend.c, created by Patrick in 2006.06.22, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_color.h"
diff -ur ./libpsd_source_original/src/gradient_fill.c ./libpsd_source/src/gradient_fill.c
--- ./libpsd_source_original/src/gradient_fill.c	2007-01-30 16:46:32.000000000 +0300
+++ ./libpsd_source/src/gradient_fill.c	2017-08-19 14:39:52.778496700 +0300
@@ -23,6 +23,7 @@
  * $Id: gradient_fill.c, created by Patrick in 2006.06.11, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -45,7 +46,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_gradient_fill));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 
 	// default value
diff -ur ./libpsd_source_original/src/gradient_map.c ./libpsd_source/src/gradient_map.c
--- ./libpsd_source_original/src/gradient_map.c	2007-01-30 16:46:38.000000000 +0300
+++ ./libpsd_source/src/gradient_map.c	2017-08-19 14:38:58.687123900 +0300
@@ -23,6 +23,7 @@
  * $Id: gradient_map.c, created by Patrick in 2006.05.22, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -30,9 +31,9 @@
 #include "psd_rect.h"
 
 
-typedef void psd_adjustment_blend_proc(psd_uint layer_info_data, psd_int * red, psd_int * green, psd_int * blue);
+typedef void psd_adjustment_blend_proc(psd_pointer layer_info_data, psd_int * red, psd_int * green, psd_int * blue);
 extern void psd_adjustment_blend_color(psd_context * context, psd_layer_record * layer, psd_rect * dst_rect, 
-	psd_adjustment_blend_proc * blend_proc, psd_uint layer_info_data);
+	psd_adjustment_blend_proc * blend_proc, psd_pointer layer_info_data);
 
 
 // Gradient settings
@@ -51,7 +52,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_gradient_map));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 
 	// Version ( =1 for Photoshop 6.0)
@@ -161,7 +162,7 @@
 	return psd_status_done;
 }
 
-void psd_layer_gradient_map_free(psd_uint info_data)
+void psd_layer_gradient_map_free(psd_pointer info_data)
 {
 	psd_layer_gradient_map * data;
 
@@ -172,7 +173,7 @@
 	psd_free(data);
 }
 
-static void psd_gradient_map_proc(psd_uint layer_info_data, psd_int * red, psd_int * green, psd_int * blue)
+static void psd_gradient_map_proc(psd_pointer layer_info_data, psd_int * red, psd_int * green, psd_int * blue)
 {
 	psd_layer_gradient_map * data = (psd_layer_gradient_map *)layer_info_data;
 	psd_int gray;
@@ -191,7 +192,7 @@
 
 psd_bool psd_layer_blend_gradient_map(psd_context * context, psd_layer_record * layer, psd_rect * dst_rect)
 {
-	psd_uint layer_info_data;
+	psd_pointer layer_info_data;
 	psd_layer_gradient_map * data;
 	psd_int i, j, start_pos, end_pos, mid_pos;
 	psd_int start_red, start_green, start_blue, 
diff -ur ./libpsd_source_original/src/gradient_overlay.c ./libpsd_source/src/gradient_overlay.c
--- ./libpsd_source_original/src/gradient_overlay.c	2007-01-30 16:46:42.000000000 +0300
+++ ./libpsd_source/src/gradient_overlay.c	2017-08-19 14:25:15.892441500 +0300
@@ -24,6 +24,7 @@
  */
 
 #include <math.h>
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/hue_saturation.c ./libpsd_source/src/hue_saturation.c
--- ./libpsd_source_original/src/hue_saturation.c	2007-01-28 16:32:30.000000000 +0300
+++ ./libpsd_source/src/hue_saturation.c	2017-08-19 14:39:52.792509700 +0300
@@ -23,6 +23,7 @@
  * $Id: hue_saturation.c, created by Patrick in 2006.05.22, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -31,9 +32,9 @@
 #include "psd_math.h"
 
 
-typedef void psd_adjustment_blend_proc(psd_uint layer_info_data, psd_int * red, psd_int * green, psd_int * blue);
+typedef void psd_adjustment_blend_proc(psd_pointer layer_info_data, psd_int * red, psd_int * green, psd_int * blue);
 extern void psd_adjustment_blend_color(psd_context * context, psd_layer_record * layer, psd_rect * dst_rect, 
-	psd_adjustment_blend_proc * blend_proc, psd_uint layer_info_data);
+	psd_adjustment_blend_proc * blend_proc, psd_pointer layer_info_data);
 
 
 // New Hue/saturation, Photoshop 5.0
@@ -49,7 +50,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_hue_saturation));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 
 	// Version ( = 2)
@@ -102,7 +103,7 @@
 	return psd_status_done;
 }
 
-static void psd_hue_saturation_proc(psd_uint layer_info_data, psd_int * red, psd_int * green, psd_int * blue)
+static void psd_hue_saturation_proc(psd_pointer layer_info_data, psd_int * red, psd_int * green, psd_int * blue)
 {
 	psd_layer_hue_saturation * data = (psd_layer_hue_saturation *)layer_info_data;
 	psd_int src_hue, src_saturation, src_lightness;
@@ -139,7 +140,7 @@
 
 psd_bool psd_layer_blend_hue_saturation(psd_context * context, psd_layer_record * layer, psd_rect * dst_rect)
 {
-	psd_uint layer_info_data;
+	psd_pointer layer_info_data;
 	psd_layer_hue_saturation * data = NULL;
 	psd_int i, j, increase, range_value[4];
 
diff -ur ./libpsd_source_original/src/image_data.c ./libpsd_source/src/image_data.c
--- ./libpsd_source_original/src/image_data.c	2007-02-04 15:02:22.000000000 +0300
+++ ./libpsd_source/src/image_data.c	2017-08-19 14:25:15.878412100 +0300
@@ -23,6 +23,7 @@
  * $Id: image_data.c, created by Patrick in 2006.05.25, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/image_resource.c ./libpsd_source/src/image_resource.c
--- ./libpsd_source_original/src/image_resource.c	2007-02-24 19:06:10.000000000 +0300
+++ ./libpsd_source/src/image_resource.c	2017-08-19 14:25:15.871405500 +0300
@@ -23,6 +23,7 @@
  * $Id: image_resource.c, created by Patrick in 2006.05.18, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/inner_glow.c ./libpsd_source/src/inner_glow.c
--- ./libpsd_source_original/src/inner_glow.c	2007-02-01 14:10:48.000000000 +0300
+++ ./libpsd_source/src/inner_glow.c	2017-08-19 14:25:15.865399200 +0300
@@ -23,6 +23,7 @@
  * $Id: inner_glow.c, created by Patrick in 2006.06.18, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/inner_shadow.c ./libpsd_source/src/inner_shadow.c
--- ./libpsd_source_original/src/inner_shadow.c	2007-02-01 14:12:16.000000000 +0300
+++ ./libpsd_source/src/inner_shadow.c	2017-08-19 14:25:15.859393300 +0300
@@ -24,6 +24,7 @@
  */
 
 #include <math.h>
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/layer_mask.c ./libpsd_source/src/layer_mask.c
--- ./libpsd_source_original/src/layer_mask.c	2007-02-06 19:34:24.000000000 +0300
+++ ./libpsd_source/src/layer_mask.c	2017-08-19 14:38:33.859902900 +0300
@@ -23,6 +23,7 @@
  * $Id: layer_mask.c, created by Patrick in 2006.05.19, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -30,9 +31,9 @@
 
 
 extern psd_status psd_get_layer_levels(psd_context * context, psd_layer_record * layer, psd_int data_length);
-extern void psd_layer_levels_free(psd_uint info_data);
+extern void psd_layer_levels_free(psd_pointer info_data);
 extern psd_status psd_get_layer_curves(psd_context * context, psd_layer_record * layer, psd_int data_length);
-extern void psd_layer_curves_free(psd_uint info_data);
+extern void psd_layer_curves_free(psd_pointer info_data);
 extern psd_status psd_get_layer_brightness_contrast(psd_context * context, psd_layer_record * layer);
 extern psd_status psd_get_layer_color_balance(psd_context * context, psd_layer_record * layer);
 extern psd_status psd_get_layer_hue_saturation(psd_context * context, psd_layer_record * layer);
@@ -43,7 +44,7 @@
 extern psd_status psd_get_layer_channel_mixer(psd_context * context, psd_layer_record * layer);
 extern psd_status psd_get_layer_photo_filter(psd_context * context, psd_layer_record * layer);
 extern psd_status psd_get_layer_gradient_map(psd_context * context, psd_layer_record * layer);
-extern void psd_layer_gradient_map_free(psd_uint info_data);
+extern void psd_layer_gradient_map_free(psd_pointer info_data);
 extern psd_status psd_get_layer_effects(psd_context * context, psd_layer_record * layer);
 extern psd_status psd_get_layer_effects2(psd_context * context, psd_layer_record * layer);
 extern psd_status psd_get_layer_solid_color(psd_context * context, psd_layer_record * layer);
@@ -53,9 +54,9 @@
 extern psd_status psd_get_layer_type_tool(psd_context * context, psd_layer_record * layer);
 extern psd_status psd_get_pattern(psd_context * context);
 extern psd_status psd_get_layer_vector_mask(psd_context * context, psd_layer_record * layer, psd_int size);
-extern void psd_layer_type_tool_free(psd_uint info_data);
+extern void psd_layer_type_tool_free(psd_pointer info_data);
 extern void psd_layer_vector_mask_free(psd_layer_record * layer);
-extern void psd_layer_effects_free(psd_uint layer_info);
+extern void psd_layer_effects_free(psd_pointer layer_info);
 extern void psd_pattern_free(psd_context * context);
 static void psd_layer_free(psd_layer_record * layer);
 
diff -ur ./libpsd_source_original/src/levels.c ./libpsd_source/src/levels.c
--- ./libpsd_source_original/src/levels.c	2007-01-28 19:16:28.000000000 +0300
+++ ./libpsd_source/src/levels.c	2017-08-19 14:40:26.109123000 +0300
@@ -23,6 +23,7 @@
  * $Id: levelsl.c, created by Patrick in 2006.05.22, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include <math.h>
 #include "libpsd.h"
 #include "psd_system.h"
@@ -52,7 +53,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_levels));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 	
 	// Version ( = 2)
@@ -133,7 +134,7 @@
 	return psd_status_done;
 }
 
-void psd_layer_levels_free(psd_uint info_data)
+void psd_layer_levels_free(psd_pointer info_data)
 {
 	psd_layer_levels * data;
 
diff -ur ./libpsd_source_original/src/outer_glow.c ./libpsd_source/src/outer_glow.c
--- ./libpsd_source_original/src/outer_glow.c	2007-02-01 14:11:28.000000000 +0300
+++ ./libpsd_source/src/outer_glow.c	2017-08-19 14:26:09.761938900 +0300
@@ -23,6 +23,7 @@
  * $Id: outer_glow.c, created by Patrick in 2006.05.23, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/path.c ./libpsd_source/src/path.c
--- ./libpsd_source_original/src/path.c	2007-02-06 19:39:16.000000000 +0300
+++ ./libpsd_source/src/path.c	2017-08-19 14:27:12.102006000 +0300
@@ -23,6 +23,7 @@
  * $Id: path.c, created by Patrick in 2007.02.06, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/pattern.c ./libpsd_source/src/pattern.c
--- ./libpsd_source_original/src/pattern.c	2007-02-04 15:59:08.000000000 +0300
+++ ./libpsd_source/src/pattern.c	2017-08-19 14:27:12.108005700 +0300
@@ -23,6 +23,7 @@
  * $Id: pattern.c, created by Patrick in 2006.06.21, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/pattern_fill.c ./libpsd_source/src/pattern_fill.c
--- ./libpsd_source_original/src/pattern_fill.c	2007-01-30 16:48:46.000000000 +0300
+++ ./libpsd_source/src/pattern_fill.c	2017-08-19 14:40:26.103116700 +0300
@@ -23,6 +23,7 @@
  * $Id: pattern_fill.c, created by Patrick in 2006.06.21, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -45,7 +46,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_pattern_fill));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 
 	// Version ( = 16 for Photoshop 6.0)
diff -ur ./libpsd_source_original/src/pattern_overlay.c ./libpsd_source/src/pattern_overlay.c
--- ./libpsd_source_original/src/pattern_overlay.c	2007-01-30 16:48:50.000000000 +0300
+++ ./libpsd_source/src/pattern_overlay.c	2017-08-19 14:27:12.084983700 +0300
@@ -24,6 +24,7 @@
  */
 
 #include <math.h>
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/photo_filter.c ./libpsd_source/src/photo_filter.c
--- ./libpsd_source_original/src/photo_filter.c	2007-01-28 15:34:06.000000000 +0300
+++ ./libpsd_source/src/photo_filter.c	2017-08-19 14:40:26.096100700 +0300
@@ -23,6 +23,7 @@
  * $Id: photo_filter.c, created by Patrick in 2006.05.22, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -41,7 +42,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_photo_filter));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 
 	// Version ( = 3)
diff -ur ./libpsd_source_original/src/posterize.c ./libpsd_source/src/posterize.c
--- ./libpsd_source_original/src/posterize.c	2007-01-28 19:14:00.000000000 +0300
+++ ./libpsd_source/src/posterize.c	2017-08-19 14:40:26.090095000 +0300
@@ -23,6 +23,7 @@
  * $Id: posterize.c, created by Patrick in 2006.05.22, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -45,7 +46,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_posterize));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 
 	// (2...255)
diff -ur ./libpsd_source_original/src/psd.c ./libpsd_source/src/psd.c
--- ./libpsd_source_original/src/psd.c	2007-02-04 15:14:48.000000000 +0300
+++ ./libpsd_source/src/psd.c	2017-08-19 14:27:12.067967700 +0300
@@ -23,6 +23,7 @@
  * $Id: psd.c, created by Patrick in 2006.05.18, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/psd_zip.c ./libpsd_source/src/psd_zip.c
--- ./libpsd_source_original/src/psd_zip.c	2007-02-05 09:20:06.000000000 +0300
+++ ./libpsd_source/src/psd_zip.c	2017-08-19 14:27:12.061962000 +0300
@@ -23,6 +23,7 @@
  * $Id: psd_zip.c, created by Patrick in 2007.02.02, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 
diff -ur ./libpsd_source_original/src/rect.c ./libpsd_source/src/rect.c
--- ./libpsd_source_original/src/rect.c	2007-01-28 15:34:52.000000000 +0300
+++ ./libpsd_source/src/rect.c	2017-08-19 14:27:12.056943600 +0300
@@ -23,6 +23,7 @@
  * $Id: rect.c, created by Patrick in 2005.06.01, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_rect.h"
 #include "psd_math.h"
diff -ur ./libpsd_source_original/src/satin.c ./libpsd_source/src/satin.c
--- ./libpsd_source_original/src/satin.c	2007-02-01 14:11:42.000000000 +0300
+++ ./libpsd_source/src/satin.c	2017-08-19 14:28:06.924964800 +0300
@@ -24,6 +24,7 @@
  */
 
 #include <math.h>
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/selective_color.c ./libpsd_source/src/selective_color.c
--- ./libpsd_source_original/src/selective_color.c	2007-01-28 17:35:04.000000000 +0300
+++ ./libpsd_source/src/selective_color.c	2017-08-19 14:40:26.123126000 +0300
@@ -23,6 +23,7 @@
  * $Id: selective_color.c, created by Patrick in 2006.05.22, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -31,9 +32,9 @@
 #include "psd_math.h"
 
 
-typedef void psd_adjustment_blend_proc(psd_uint layer_info_data, psd_int * red, psd_int * green, psd_int * blue);
+typedef void psd_adjustment_blend_proc(psd_pointer layer_info_data, psd_int * red, psd_int * green, psd_int * blue);
 extern void psd_adjustment_blend_color(psd_context * context, psd_layer_record * layer, psd_rect * dst_rect, 
-	psd_adjustment_blend_proc * blend_proc, psd_uint layer_info_data);
+	psd_adjustment_blend_proc * blend_proc, psd_pointer layer_info_data);
 
 
 psd_status psd_get_layer_selective_color(psd_context * context, psd_layer_record * layer)
@@ -48,7 +49,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_selective_color));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 
 	// Version ( = 1)
@@ -86,7 +87,7 @@
 	return psd_status_done;
 }
 
-static void psd_selective_color_proc(psd_uint layer_info_data, psd_int * red, psd_int * green, psd_int * blue)
+static void psd_selective_color_proc(psd_pointer layer_info_data, psd_int * red, psd_int * green, psd_int * blue)
 {
 	psd_layer_selective_color * data = (psd_layer_selective_color *)layer_info_data;
 	psd_int hue, saturation, lightness;
@@ -186,7 +187,7 @@
 
 psd_bool psd_layer_blend_selective_color(psd_context * context, psd_layer_record * layer, psd_rect * dst_rect)
 {
-	psd_uint layer_info_data;
+	psd_pointer layer_info_data;
 	psd_layer_selective_color * data = NULL;
 	psd_int i;
 
diff -ur ./libpsd_source_original/src/solid_color.c ./libpsd_source/src/solid_color.c
--- ./libpsd_source_original/src/solid_color.c	2007-01-30 10:23:42.000000000 +0300
+++ ./libpsd_source/src/solid_color.c	2017-08-19 14:40:26.082087400 +0300
@@ -23,6 +23,7 @@
  * $Id: solid_color.c, created by Patrick in 2006.05.24, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -44,7 +45,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_solid_color));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 
 	// Version ( = 16 for Photoshop 6.0)
diff -ur ./libpsd_source_original/src/stream.c ./libpsd_source/src/stream.c
--- ./libpsd_source_original/src/stream.c	2007-02-01 14:20:42.000000000 +0300
+++ ./libpsd_source/src/stream.c	2017-08-19 14:27:46.283121400 +0300
@@ -23,6 +23,7 @@
  * $Id: stream.c, created by Patrick in 2006.05.18, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
diff -ur ./libpsd_source_original/src/stroke.c ./libpsd_source/src/stroke.c
--- ./libpsd_source_original/src/stroke.c	2007-01-30 16:49:48.000000000 +0300
+++ ./libpsd_source/src/stroke.c	2017-08-19 14:27:46.277116000 +0300
@@ -24,6 +24,7 @@
  */
 
 #include <math.h>
+#include <string.h>
 #include "libpsd.h"
 #include "psd_config.h"
 #include "psd_system.h"
diff -ur ./libpsd_source_original/src/threshold.c ./libpsd_source/src/threshold.c
--- ./libpsd_source_original/src/threshold.c	2007-01-28 15:36:30.000000000 +0300
+++ ./libpsd_source/src/threshold.c	2017-08-19 14:40:26.116119700 +0300
@@ -23,6 +23,7 @@
  * $Id: threshold.c, created by Patrick in 2006.05.22, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -45,7 +46,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_threshold));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 
 	// (1...255)
@@ -62,7 +63,7 @@
 
 psd_bool psd_layer_blend_threshold(psd_context * context, psd_layer_record * layer, psd_rect * dst_rect)
 {
-	psd_uint layer_info_data;
+	psd_pointer layer_info_data;
 	psd_layer_threshold * data = NULL;
 	psd_int i;
 	psd_uchar lookup_table[256];
diff -ur ./libpsd_source_original/src/type_tool.c ./libpsd_source/src/type_tool.c
--- ./libpsd_source_original/src/type_tool.c	2007-01-30 16:50:10.000000000 +0300
+++ ./libpsd_source/src/type_tool.c	2017-08-19 14:36:01.145047600 +0300
@@ -23,6 +23,7 @@
  * $Id: type_tool.c, created by Patrick in 2006.05.30, libpsd@graphest.com Exp $
  */
 
+#include <string.h>
 #include "libpsd.h"
 #include "psd_system.h"
 #include "psd_stream.h"
@@ -41,7 +42,7 @@
 	if(data == NULL)
 		return psd_status_malloc_failed;
 	memset(data, 0, sizeof(psd_layer_type_tool));
-	layer->layer_info_data[layer->layer_info_count] = (psd_uint)data;
+	layer->layer_info_data[layer->layer_info_count] = (psd_pointer)data;
 	layer->layer_info_count ++;
 
 	// Version ( = 1)
@@ -205,7 +206,7 @@
 	return psd_status_done;
 }
 
-void psd_layer_type_tool_free(psd_uint info_data)
+void psd_layer_type_tool_free(psd_pointer info_data)
 {
 	psd_layer_type_tool * data;
 	psd_int i;
