--- "lpng1625_original\\scripts\\genout.cmake.in"	2016-05-04 10:34:50.000000000 +0300
+++ "lpng1625\\scripts\\genout.cmake.in"	2016-09-16 11:24:10.171056100 +0300
@@ -34,6 +34,13 @@
   endif()
 endif()
 
+# PATCHED: VS ARM Compiler requires _ARM_WINAPI_PARTITION_DESKTOP_SDK_AVAILABLE to be set
+if(WIN32)
+  if (CMAKE_C_COMPILER MATCHES "VC/bin/x86_arm" OR CMAKE_C_COMPILER MATCHES "VC/bin/amd64_arm")
+    set(PLATFORM_C_FLAGS ${PLATFORM_C_FLAGS} /D "_ARM_WINAPI_PARTITION_DESKTOP_SDK_AVAILABLE=1")
+  endif()
+endif()
+
 get_filename_component(INPUTEXT "${INPUT}" EXT)
 get_filename_component(OUTPUTEXT "${OUTPUT}" EXT)
 get_filename_component(INPUTBASE "${INPUT}" NAME_WE)
